src/mixins/raster-layer-poly-mixin.js

```diff
@@ -624,33 +690,38 @@ export default function rasterLayerPolyMixin(_layer) {
 
   _layer.viewBoxDim = createRasterLayerGetterSetter(_layer, null)
 
-  _layer._genVega = function(chart, layerName, group) {
-    const mapBounds = chart.map().getBounds()
-
-    const columnExpr = `${_layer.getState().encoding.geoTable}.${
-      _layer.getState().encoding.geocol
-    }`
-
-    const bboxFilter = `ST_XMax(${columnExpr}) >= ${mapBounds._sw.lng} AND ST_XMin(${columnExpr}) <= ${mapBounds._ne.lng} AND ST_YMax(${columnExpr}) >= ${mapBounds._sw.lat} AND ST_YMin(${columnExpr}) <= ${mapBounds._ne.lat}`
-
-    const allFilters = _layer.crossfilter().getFilter(layerName)
-    const otherChartFilters = allFilters.filter(
-      (f, i) =>
-        i !== _layer.dimension().getDimensionIndex() && f !== "" && f != null
-    )
-
+  _layer._genVega = function(chart, layerName) {
     let polyFilterString = ""
-    let firstElem = true
+    let bboxFilter = ""
+    if (!isContourType(_layer.getState())) {
```

Comment:
Dramatic diff... this just doesn't do the bounding box stuff if its a contour chart, wraps it in `if (!isContourType(_layer.getState())) {`

Reply:
Also separated some logic into `getPolygonScale` a separate function above here